src/wallet/wallet.cpp

```diff
@@ -1064,14 +1083,15 @@ bool CWallet::AddToWalletIfInvolvingMe(const CTransaction& tx, const uint256& bl
         if (fExisted && !fUpdate) return false;
 
         // Check tx for Sapling notes
-        Optional<mapSaplingNoteData_t> saplingNoteData {nullopt};
+        mapSaplingNoteData_t saplingNoteDataMap;
```

Comment:
The goal here for having it as optional was to not initialize the map at all if it's not needed. Below, we could just ask `if (HasSaplingSPKM() && tx.IsShielded())`, so it doesn't need to even try to get into the function.

Reply:
yep, done